### ISSUE TITLE
Various Winsock constants missing

### ISSUE THREAD

**Issue by kennykerr** [microsoft/win32metadata#844]
Constants like `SIO_UDP_CONNRESET`, `SIO_BSP_HANDLE`, `SIO_BASE_HANDLE` etc. are missing. 

They appear to come from `mswsock.h`.

https://github.com/tokio-rs/mio/pull/1556#discussion_r831173927

**Comment by kennykerr** [comment]
FYI IO control codes are unsigned.

**Comment by riverar** [comment]
@kennykerr Thanks, fixed.

**Comment by mikebattista** [comment]
Need to automate the below pattern in the constant scraper and then recognize constants defined with these macros.

```
/*
 * WinSock 2 extension -- manifest constants for WSAIoctl()
 */
#define IOC_UNIX                      0x00000000
#define IOC_WS2                       0x08000000
#define IOC_PROTOCOL                  0x10000000
#define IOC_VENDOR                    0x18000000

#if (_WIN32_WINNT >= 0x0600)
/*
 * WSK-specific IO control codes are Winsock2 codes with the highest-order
 * 3 bits of the Vendor/AddressFamily-specific field set to 1.
 */
#define IOC_WSK                       (IOC_WS2|0x07000000)
#endif //(_WIN32_WINNT >= 0x0600)

#define _WSAIO(x,y)                   (IOC_VOID|(x)|(y))
#define _WSAIOR(x,y)                  (IOC_OUT|(x)|(y))
#define _WSAIOW(x,y)                  (IOC_IN|(x)|(y))
#define _WSAIORW(x,y)                 (IOC_INOUT|(x)|(y))
```

**Comment by mikebattista** [comment]
Will fix #671 as well.
